### PR TITLE
Add extensive metadata guides

### PR DESCRIPTION
**Pull request by cbetta**
Added new metadata guides.

### CODE REVIEW

File: content/guides/metadata/0-index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+## Templates, Fields, Instances, and Cascade Policies
+
+Working with metadata requires a developer to work with a few
+distinct types of resources.
+
+* **Templates:**  A [metadata template][template] describes a re-usable set of
+  key/value pairs that can be assigned to a file. For example, an `invoiceData`
+  template might hold data about an invoice, having a field for the invoice ID
+  as well as the customer ID.
+  * **Fields:**  A [metadata template field][field] describes a specific
```
Comment (PreciselyAlyss) [inline content/guides/metadata/0-index.md]: Is the extra spacing intended here for formatting?

File: content/guides/metadata/1-templates/0-index.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Metadata templates are grouped into two distinct groups, or **scopes**.
+
+<CTA to='g://metadata/templates/scopes'>
+  Learn more about Metadata Scopes
+</CTA>
+
+## Restrictions
+
+There is a limit of 500 templates per enterprise.
```
Comment (PreciselyAlyss) [inline content/guides/metadata/1-templates/0-index.md]: Do we need an extra callout here for who/the required permissions to make templates?

File: content/guides/metadata/0-index.md (hunk 1 above)
Comment (cbetta) [reply content/guides/metadata/0-index.md]: The extra left indent? Yes, cause fields are part of templates.

File: content/guides/metadata/1-templates/0-index.md (hunk 2 above)
Comment (cbetta) [reply content/guides/metadata/1-templates/0-index.md]: Good point, I'll do

**Pull request merged**